flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -362,30 +358,35 @@ private void testSyncSavepointWithEndInput(
                 "savepointResult");
         harness.processAll();
 
-        Assert.assertEquals(expectEndInput, TestBoundedOneInputStreamOperator.isInputEnded());
+        assertThat(TestBoundedOneInputStreamOperator.isInputEnded()).isEqualTo(expectEndInput);
     }
 
     @Test
-    public void testCleanUpExceptionSuppressing() throws Exception {
+    void testCleanUpExceptionSuppressing() throws Exception {
         try (StreamTaskMailboxTestHarness<String> testHarness =
                 new StreamTaskMailboxTestHarnessBuilder<>(OneInputStreamTask::new, STRING_TYPE_INFO)
                         .addInput(STRING_TYPE_INFO)
                         .setupOutputForSingletonOperatorChain(new FailingTwiceOperator())
                         .build()) {
 
-            try {
-                testHarness.processElement(new StreamRecord<>("Doesn't matter", 0));
-                throw new RuntimeException("Expected an exception but ran successfully");
-            } catch (Exception ex) {
-                ExceptionUtils.assertThrowable(ex, ExpectedTestException.class);
-            }
+            assertThatThrownBy(
+                            () ->
+                                    testHarness.processElement(
+                                            new StreamRecord<>("Doesn't matter", 0)))
+                    .satisfies(
+                            (Consumer<Throwable>)
+                                    throwable ->
+                                            ExceptionUtils.assertThrowable(
+                                                    throwable, ExpectedTestException.class));
 
-            try {
-                testHarness.finishProcessing();
-            } catch (Exception ex) {
-                // todo: checking for suppression if there are more exceptions during cleanup
-                ExceptionUtils.assertThrowable(ex, FailingTwiceOperator.CloseException.class);
-            }
+            // todo: checking for suppression if there are more exceptions during cleanup
+            assertThatThrownBy(testHarness::finishProcessing)
+                    .satisfies(
+                            (ThrowingConsumer<Throwable>)
+                                    throwable ->
+                                            ExceptionUtils.assertThrowable(
+                                                    throwable,
+                                                    FailingTwiceOperator.CloseException.class));
```

Comment:
```suggestion
                    .hasCause(FailingTwiceOperator.CloseException.class));
```

Can it work?

Reply:
<img width="1233" alt="image" src="https://user-images.githubusercontent.com/64569824/221202347-41aeb24a-2324-46cb-9985-0a3e5595faf6.png">

So, let's keep the current lines ?

Reply:
```suggestion
                    .isInstanceOf(FailingTwiceOperator.CloseException.class);
```

I have tried it, it works.

Reply:
Yes, the line is also my original migration line.
@1996fanrui 
@reswqa described the reason why we still need the  `ExceptionUtils.assertThrowable` at  https://github.com/apache/flink/pull/21999#discussion_r1116493534 due to the depth of the exception.

Reply:
Hi @RocMarshal , thanks for your clarification, I got it.

Hi @reswqa , I found `FlinkAssertions.anyCauseMatches` is similar with `ExceptionUtils.assertThrowable`, and it can find the throwable from more deeper level, and it's better than `ExceptionUtils.assertThrowable`.

What do you think?


```suggestion
                    .satisfies(
                            (ThrowingConsumer<Throwable>)
                                    throwable ->
                                            ExceptionUtils.assertThrowable(
                                                    throwable,
                                                    FailingTwiceOperator.CloseException.class));
```

Reply:
Good suggestion @1996fanrui, Make sense to me.